src/client/index.ts

```diff
@@ -18,98 +49,228 @@ export interface NotificationOptions {
+ * interacted with, and what metadata was attached to the action.
+ *
+ * See {@link Actions} for a full example of how actions are defined, and how an application can listen to and handle
+ * them.
+ *
+ * This type includes a generic type argument, should applications wish to define their own interface for action
+ * results. See {@link NotificationActionResult} for details.
  *
  * @event
  */
-export interface NotificationClickedEvent {
-    type: 'notification-clicked';
-    notification: Notification;
+export interface NotificationActionEvent<T = {}> {
+    type: 'notification-action';
+
+    /**
+     * The notification that created this action
+     */
+    notification: Readonly<Notification>;
+
+    /**
+     * Indicates what triggered this action:
+     *
+     * - **control**: The user interacted with one of the controls within the notification. This currently means a button
+     * click, but other control types will be added in future releases.
+     * - **body**: The user clicked the body of the notification itself. Any clicks of the notification that don't hit a
+     * control will result in this event being fired.
+     * - **programmatic**: The action was triggered programmatically by an application (*Not currently supported*).
+     */
+    trigger: 'control'|'body'|'programmatic';
+
+    /**
+     * The control whose interaction resulted in this action being raised. Will only be present when {@link trigger} is
+     * `control`.
+     *
+     * Future versions of the service will add additional controls beyond buttons, and interactions with these new
+     * control types will also come through the same `notification-action` event. For best forward-compatibility,
+     * assume that `control` can be optional, and may have a `type` property of something other than `'button'`.
+     *
+     * This field is marked optional as future versions of the service will also include alternate methods of raising
+     * `notification-action` events that do not originate from a button or other control.
+     *
+     * When `control` is present, this object will always be strictly equal to one of the control definitions within
+     * `notification`. This means `indexOf` checks and other equality checks can be performed on this field if
+     * required, such as:
+     *
+     * ```ts
+     * function onNotificationAction(event: NotificationActionEvent): void {
+     *     if (event.control && event.control.type === 'button') {
+     *         const butttonIndex = event.notification.buttons.indexOf(event.control);
+     *
+     *         // Handle button click
+     *         // . . .
+     *     }
+     * }
+     * ```
+     */
+    control?: ControlOptions;
+
+    /**
+     * Application-defined metadata that this event is passing back to the application.
+     *
+     * A `notification-action` event is only raised if the {@link NotificationOptions|notification definition} includes
```

Comment:
Should be `{@link NotificationOptions} notification definition`

Reply:
Not sure I follow, that doesn't seem very readable?..

Reply:
The link has `NotificationOptions|notification definition` in it. I assume:
> A `notification-action` event is only raised if the {@link NotificationOptions|notification definition} includes control will result in this event being fired.
Should be:
> A `notification-action` event is only raised if the {@link NotificationOptions} definition includes control will result in this event being fired.

Reply:
Actually reading the whole sentence again it just doesn't make sense.

> A `notification-action` event is only fired when a notification {@link NotificationOptions} including a control is interacted with.

Reply:
Forgot that `{@link Type | text link}` turns into a hyper link with the text after the `|`, so ignore above.

Still this sentence doesn't make sense:

>A `notification-action` event is only raised if the notification definition **includes control will result in this event being fired**.

Reply:
Not sure where you're getting that from... I see:

> A `notification-action` event is only raised if the {@link NotificationOptions|notification definition} includes a payload on the actions of one or more controls (and if the user then performs the corresponding action).

Still slightly clunky, but that's the best wording I could come up with at the time - open to suggestions.